### PR TITLE
find_files() added features, file components and associated yaml files.

### PR DESCRIPTION
**Pull request by berland**
Feature to be used in #49

### CODE REVIEW

File: tests/test_realization.py [hunk 1]
```diff
@@ -815,3 +816,88 @@ def test_drop():
...
+    real = ensemble.ScratchRealization(realdir)
+
+    # Setup example files with some yaml data:
+    findable_files = ["grid1.gri", "grid2.gri"]
+    for filename in findable_files:
+        with open(os.path.join(realdir, filename), "w") as fileh:
+            fileh.write("baah")
+        yamlfile = "." + filename + ".yml"
+        with open(os.path.join(realdir, yamlfile), "w") as fileh:
+            fileh.write(yaml.dump(dict(a=dict(x=1, y=2), b="bar")))
```
Comment (fahaddilib) [inline tests/test_realization.py]: Could you give an example of the use case of having a yaml files with key-values pairs? Not sure from the test how this yaml file would relates to the *gri files?

Comment (berland) [reply tests/test_realization.py]: F.ex. *.gri files (surfaces) can have an arbitrary amount of meta-information attached to them, hints for visualization, coordinates system, dates, source, etc etc. 

The functionality for  flattening yaml-dict structures to key-value is in order to include the values as tabular data when you ask an ensemble object for its `.files`. An application can have its own selection logic based on the meta-data for which surfaces to display and how.

Comment (berland) [reply tests/test_realization.py]: See https://github.com/equinor/fmu-ensemble/blob/rmrc/tests/test_rmrc.py#L48 for example of usage.

Comment (fahaddilib) [reply tests/test_realization.py]: Make sense and the another implementation is if the meta-information is directly in the filename denoted by --

**Pull request merged**